commit f677d7702d48b7b3dfcce3b2c0db601dbee0aa24
Author: Tulio Magno Quites Machado Filho <tuliom@gmail.com>
Date:   Sun Jan 25 23:54:25 2009 +0100

    ath5k: support LED's on emachines E510 notebook
    
    Add vendor ID for AMBIT and use it to set the ath5k LED gpio.
    
    base.c:
    Changes-licensed-under: 3-Clause-BSD
    
    Signed-off-by: Tulio Magno Quites Machado Filho <tuliom@gmail.com>
    Acked-by: Bob Copeland <me@bobcopeland.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ath5k/base.c b/drivers/net/wireless/ath5k/base.c
index b3f41ac..368db94 100644
--- a/drivers/net/wireless/ath5k/base.c
+++ b/drivers/net/wireless/ath5k/base.c
@@ -2626,8 +2626,12 @@ ath5k_init_leds(struct ath5k_softc *sc)
 		sc->led_pin = 1;
 		sc->led_on = 1;  /* active high */
 	}
-	/* Pin 3 on Foxconn chips used in Acer Aspire One (0x105b:e008) */
-	if (pdev->subsystem_vendor == PCI_VENDOR_ID_FOXCONN) {
+	/*
+	 * Pin 3 on Foxconn chips used in Acer Aspire One (0x105b:e008) and
+	 * in emachines notebooks with AMBIT subsystem.
+	 */
+	if (pdev->subsystem_vendor == PCI_VENDOR_ID_FOXCONN ||
+	    pdev->subsystem_vendor == PCI_VENDOR_ID_AMBIT) {
 		__set_bit(ATH_STAT_LEDSOFT, sc->status);
 		sc->led_pin = 3;
 		sc->led_on = 0;  /* active low */
diff --git a/include/linux/pci_ids.h b/include/linux/pci_ids.h
index 5b7a48c..b7697c9 100644
--- a/include/linux/pci_ids.h
+++ b/include/linux/pci_ids.h
@@ -1967,6 +1967,8 @@
 
 #define PCI_VENDOR_ID_SAMSUNG		0x144d
 
+#define PCI_VENDOR_ID_AMBIT		0x1468
+
 #define PCI_VENDOR_ID_MYRICOM		0x14c1
 
 #define PCI_VENDOR_ID_TITAN		0x14D2
